Automattic/woocommerce-payments
# Implement pending account status

Fixes #5672 

### Changes proposed in this Pull Request

* Support `Pending` status in the `StatusChip` component.
* Support tooltip in the `Chip` component.
* Support pending states for `DepositsStatus` and `PaymentsStatus` components.
* Apply pending status for all the components listed above is `accountStatus.status` is `pending_verification`.
* Update tests accordingly.

### Testing instructions

#### Pre-requisites
* Checkout to `add/5672-support-pending-account-status`
* Change client version to `5.9.0` in https://github.com/Automattic/woocommerce-payments/blob/4f10067f81222300ac480791a7ebc7d822a62a56/woocommerce-payments.php#L15-L15

#### Local testing
* Follow testing instructions from server PR (Local testing)
* Onboard new account (non-progressive). You can pick estimated timeline as `6+ months` for Full KYC
* You should see this state of account details 
![image](https://user-images.githubusercontent.com/79862886/234512851-55b00543-343c-40b6-b0f1-5fb35600630c.png)
* Hover on `Pending` status and verify you can see the tooltip 
![image](https://user-images.githubusercontent.com/79862886/234513716-c4e69096-578b-4e69-85e3-119bdc7fe43e.png)
* Roll-back changes from the server PR, clear the account cache.
* You should see that account is complete now 
![image](https://user-images.githubusercontent.com/79862886/234514466-5b135775-f419-407c-a358-82eaeff38dcc.png)

#### Sandbox testing

* Follow testing instructions from server PR (Sandbox testing)
* Onboard new account (non-progressive). You can pick estimated timeline as 6+ months for Full KYC
* You should see this state of account details 
![image](https://user-images.githubusercontent.com/79862886/234512851-55b00543-343c-40b6-b0f1-5fb35600630c.png)
* Hover on `Pending` status and verify you can see the tooltip 
![image](https://user-images.githubusercontent.com/79862886/234513716-c4e69096-578b-4e69-85e3-119bdc7fe43e.png)
* Wait a few mins, clear the cache and reload the page, now it should be complete 
![image](https://user-images.githubusercontent.com/79862886/234514466-5b135775-f419-407c-a358-82eaeff38dcc.png)

#### Bonus
* You could test that for client version less than `5.9.0` pending status is overridden to `Restricted` (as it was before this PR)
-------------------


**Post merge**
